bin/start-collectd

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+: ${GRAPHITE_HOST:=}
+: ${GRAPHITE_PORT:="2003"}
+
+set -eu
+
+CONF_TEMPLATE=/etc/collectd/collectd.conf.tpl
+CONF_DEST=/etc/collectd/collectd.conf
+
+HOSTNAME=$(wget -O - --tries 1 --timeout 0.2 http://169.254.169.254/1.0/meta-data/instance-id 2>/dev/null || hostname)
+
+if [ -n "$GRAPHITE_HOST" ]; then
+    sed -e "s/{{ hostname }}/$HOSTNAME/g" \
```

Comment:
I have a suggestion which might make this a bit nicer. Instead of doing this hacky thing like I did for nginx, perhaps we could use an

```
<Include "/etc/collectd/collectd.conf.d">
  Filter "*.conf"
</Include>
```

And then rather than using sed, do something like:

```shell
if [ -n "$GRAPHITE_HOST" ]; then
  cat >/etc/collectd/collectd.conf.d/graphite.conf <<EOM
<Plugin write_graphite>
	<Node "default">
		Host "${GRAPHITE_HOST}"
		Port "${GRAPHITE_PORT}"
		Protocol "tcp"
		LogSendErrors true
		Prefix "collectd."
		StoreRates true
		AlwaysAppendDS false
		EscapeCharacter "_"
	</Node>
</Plugin>
EOM
fi
```

Of course, you'll need to tweak the bits that you `chown` in the Dockerfile to be the `.d` directory rather than the root config file, but I think that'll work...

In retrospect I probably could have and should have done something like this for nginx.